packages/ui/src/theme/css/component/radio.scss

```diff
@@ -0,0 +1,80 @@
+.amplify-radio {
+  gap: inherit;
+}
+
+.amplify-radio__button {
+  width: var(--amplify-components-radio-button-width);
+  height: var(--amplify-components-radio-button-height);
+  box-sizing: var(--amplify-components-radio-button-box-sizing);
+  border-width: var(--amplify-components-radio-button-border-width);
+  border-style: var(--amplify-components-radio-button-border-style);
+  border-radius: var(--amplify-components-radio-button-border-radius);
+  border-color: var(--amplify-components-radio-button-border-color);
+  color: var(--amplify-components-radio-button-color);
+  background-color: var(--amplify-components-radio-button-background-color);
+  transition-property: var(
+    --amplify-components-radio-button-transition-property
+  );
+  transition-duration: var(
+    --amplify-components-radio-button-transition-duration
+  );
+
+  // For case where box shadow doesn't work
+  // https://stackoverflow.com/questions/65663864/does-replacing-a-form-inputs-outline-with-box-shadow-violate-any-wcag-guideline
+  outline-color: var(--amplify-components-radio-button-outline-color);
+  outline-style: var(--amplify-components-radio-button-outline-style);
+  outline-width: var(--amplify-components-radio-button-outline-width);
+  outline-offset: var(--amplify-components-radio-button-outline-offset);
+
+  &::before {
+    content: var(--amplify-components-radio-button-before-content);
+    display: var(--amplify-components-radio-button-before-display);
+    width: var(--amplify-components-radio-button-before-width);
+    height: var(--amplify-components-radio-button-before-height);
+    border-radius: var(--amplify-components-radio-button-before-border-radius);
+    background-color: var(
+      --amplify-components-radio-button-before-background-color
+    );
+  }
+
+  &[data-size='small'] {
+    width: var(--amplify-components-radio-button-small-width);
+    height: var(--amplify-components-radio-button-small-height);
+  }
+  &[data-size='large'] {
+    width: var(--amplify-components-radio-button-large-width);
+    height: var(--amplify-components-radio-button-large-height);
+  }
+}
+
+.amplify-radio__input:checked + .amplify-radio__button {
```

Comment:
Sibling selectors are likely to break if any refactoring happens on the HTML. A data attribute selector (`data-checked`) on the button element would probably be less brittle.

Reply:
Mentioned above